src/BlazorWebView/tests/MauiDeviceTests/Elements/BlazorWebViewTests.cs

```diff
@@ -129,11 +129,120 @@ public async Task BlazorWebViewUsesStartPath()
 				var platformWebView = bwvHandler.PlatformView;
 				await WebViewHelpers.WaitForWebViewReady(platformWebView);
 
-				// Wait for the no-op component to load
+				// Wait for the component to load
 				await WebViewHelpers.WaitForControlDiv(bwvHandler.PlatformView, controlValueToWaitFor: "Found the start path with: 'SomeData'");
 			});
 		}
 
+		[Fact]
+		public async Task BlazorWebViewDispatchGetsScopedServices()
+		{
+			EnsureHandlerCreated(additionalCreationActions: appBuilder =>
+			{
+				appBuilder.Services.AddMauiBlazorWebView();
+				appBuilder.Services.AddScoped<ExampleJsInterop>();
+			});
+
+			var bwv = new BlazorWebViewWithCustomFiles
+			{
+				HostPage = "wwwroot/index.html",
+				CustomFiles = new Dictionary<string, string>
+				{
+					{ "index.html", TestStaticFilesContents.DefaultMauiIndexHtmlContent },
+				},
+			};
+			bwv.RootComponents.Add(new RootComponent { ComponentType = typeof(NoOpComponent), Selector = "#app", });
+
+			await InvokeOnMainThreadAsync(async () =>
+			{
+				var bwvHandler = CreateHandler<BlazorWebViewHandler>(bwv);
+				var platformWebView = bwvHandler.PlatformView;
+				await WebViewHelpers.WaitForWebViewReady(platformWebView);
+
+				// Wait for the no-op component to load
+				await WebViewHelpers.WaitForControlDiv(bwvHandler.PlatformView, controlValueToWaitFor: "Static");
+
+				// Use BlazorWebView.TryDispatchAsync to access scoped services
+				var calledWorkItem = await bwv.TryDispatchAsync(async services =>
+				{
+					var jsInterop = services.GetRequiredService<ExampleJsInterop>();
+					await jsInterop.UpdateControlDiv("some new value");
+				});
+
+				Assert.True(calledWorkItem);
+
+				// Wait for the no-op component to show the new value
+				await WebViewHelpers.WaitForControlDiv(bwvHandler.PlatformView, controlValueToWaitFor: "some new value");
+			});
+		}
+
+		[Fact]
+		public async Task BlazorWebViewDispatchBeforeRunningReturnsFalse()
+		{
+			EnsureHandlerCreated(additionalCreationActions: appBuilder =>
+			{
+				appBuilder.Services.AddMauiBlazorWebView();
+				appBuilder.Services.AddScoped<ExampleJsInterop>();
+			});
+
+			var bwv = new BlazorWebViewWithCustomFiles
+			{
+				HostPage = "wwwroot/index.html",
+				CustomFiles = new Dictionary<string, string>
+				{
+					{ "index.html", TestStaticFilesContents.DefaultMauiIndexHtmlContent },
+				},
+			};
+			bwv.RootComponents.Add(new RootComponent { ComponentType = typeof(NoOpComponent), Selector = "#app", });
+
+			await InvokeOnMainThreadAsync(async () =>
+			{
+				// Try to dispatch before the MAUI handler is created
+				var calledWorkItem = await bwv.TryDispatchAsync(_ => throw new NotImplementedException());
```

Comment:
I added one test just to show that we do sometimes return `false` (meaning the dispatch did not go through). In this case it's the most trivial "fail" case because the BlazorWebView wasn't even started at all.

There are some far more complex "fail" cases that I couldn't figure out how to test, because they're mostly to guard against "race conditions" where a navigation happens exactly during the dispatch. One is better than none.